packages/utils/src/totals/cart/index.ts

```diff
@@ -0,0 +1,222 @@
+import { BigNumberInput, CartLikeWithTotals } from "@medusajs/types"
+import { BigNumber } from "../big-number"
+import { GetItemTotalInput, getLineItemsTotals } from "../line-item"
+import { MathBN } from "../math"
+import {
+  GetShippingMethodTotalInput,
+  getShippingMethodsTotals,
+} from "../shipping-method"
+import { transformPropertiesToBigNumber } from "../transform-properties-to-bignumber"
+
+interface TotalsConfig {
+  includeTaxes?: boolean
+}
+
+export interface DecorateCartLikeInputDTO {
+  items?: {
+    id?: string
+    unit_price: BigNumberInput
+    quantity: BigNumberInput
+    adjustments?: { amount: BigNumberInput }[]
+    tax_lines?: {
+      rate: BigNumberInput
+      is_tax_inclusive?: boolean
+    }[]
+  }[]
+  shipping_methods?: {
+    id?: string
+    amount: BigNumberInput
+    adjustments?: { amount: BigNumberInput }[]
+    tax_lines?: {
+      rate: BigNumberInput
+      is_tax_inclusive?: boolean
+    }[]
+  }[]
+  region?: {
+    automatic_taxes?: boolean
+  }
+}
+
+export function decorateCartTotals(
+  cartLike: DecorateCartLikeInputDTO,
+  config: TotalsConfig = {}
+): CartLikeWithTotals {
+  transformPropertiesToBigNumber(cartLike)
```

Comment:
as discussed, we might have lost precision when serializing, I am mentionning it here just so that we remember about it until we find the right approach :D

Reply:
will add tests to that in another PR where more modules will be using BigNumber